### PR TITLE
feature/571_add_batching_support_to_sth

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #571 
* 100% unit tests passed:
```
Tests run: 80, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
time=2016-02-10T17:01:13.589CET | lvl=INFO | trans=1455119397-642-0000000433 | svc=chapterdata | subsvc=demo | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSTHSink[160] : [sth-sink] Persisting data at OrionSTHSink. Database: sth_chapterdata, Collection: sth_/demo_room1_room_temperature.aggr, Data: 1455119425,2016-02-10T15:50:25.677Z,Room1,Room,temperature,Room,433,[{"name":"resolution","type":"integer","value":"216.50"}]
time=2016-02-10T17:01:14.311CET | lvl=INFO | trans=1455119397-642-0000000433 | svc=chapterdata | subsvc=demo | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSTHSink[160] : [sth-sink] Persisting data at OrionSTHSink. Database: sth_chapterdata, Collection: sth_/demo_room1_room_pressure.aggr, Data: 1455119425,2016-02-10T15:50:25.677Z,Room1,Room,pressure,Room,433,[{"name":"resolution","type":"integer","value":"216.50"}]
time=2016-02-10T17:01:14.984CET | lvl=INFO | trans=1455119397-642-0000000433 | svc=chapterdata | subsvc=demo | function=processOneByOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[257] : Finishing transaction (1455119397-642-0000000433)
time=2016-02-10T17:01:15.028CET | lvl=INFO | trans=1455119397-642-0000000434 | svc=chapterdata | subsvc=demo | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSTHSink[160] : [sth-sink] Persisting data at OrionSTHSink. Database: sth_chapterdata, Collection: sth_/demo_room1_room_temperature.aggr, Data: 1455119425,2016-02-10T15:50:25.719Z,Room1,Room,temperature,Room,434,[{"name":"resolution","type":"integer","value":"217.00"}]
time=2016-02-10T17:01:15.815CET | lvl=INFO | trans=1455119397-642-0000000434 | svc=chapterdata | subsvc=demo | function=persistOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSTHSink[160] : [sth-sink] Persisting data at OrionSTHSink. Database: sth_chapterdata, Collection: sth_/demo_room1_room_pressure.aggr, Data: 1455119425,2016-02-10T15:50:25.719Z,Room1,Room,pressure,Room,434,[{"name":"resolution","type":"integer","value":"217.00"}]
time=2016-02-10T17:01:16.506CET | lvl=INFO | trans=1455119397-642-0000000434 | svc=chapterdata | subsvc=demo | function=processOneByOne | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[257] : Finishing transaction (1455119397-642-0000000434)
...
...
> show collections
sth_/demo_room1_room_pressure.aggr
sth_/demo_room1_room_temperature.aggr
```
* Assignee @pcoello25

### CODE REVIEW

File: doc/flume_extensions_catalogue/orion_sth_sink.md [hunk 1]
```diff
@@ -248,24 +253,33 @@ In case of using hashes as part of the collection names and to let the user or d
...
+The problem with such an approach (updates versus inserts) is there is no operation in the Mongo API enabling the update of a batch. As much, there exists a `updateMany` operation, but it is about updating many collections with a single data (the updated collections are those matching the given query).
+
+Thus, `OrionSTHSink` does not implement a real batching mechanism as usual. Please observe the batching accumulation is still valid, since many events may be accumulated and processed at the same time, even in the case of configuring a batch size of 1, a single notification may include several context elements. The difference with regard to the other sinks is the events within the batch will be processed one by one after all.
+
+[Top](#top)
+
 ##<a name="section4"></a>Programmers guide
 ###<a name="section4.1"></a>`OrionSTHSink` class
-`OrionSTHSink` extends `OrionMongoBaseSink`, which as any other NGSI-like sink extends the base `OrionSink`. The methods that are extended are by `OrionMongoBaseSink` are:
+`OrionSTHSink` extends `OrionMongoBaseSink`, which s any other NGSI-like sink, extends the base `OrionSink`. The methods that are extended are:
```
Comment (pcoello25) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: Missing letter in `which s any other NGSI-like sink` :)

File: doc/flume_extensions_catalogue/orion_sth_sink.md [hunk 2]
```diff
@@ -248,24 +253,33 @@ In case of using hashes as part of the collection names and to let the user or d
...
 
+    void persistBatch(Batch batch) throws Exception;
+    
+A `Batch` contanins a set of `CygnusEvent` objects, which are the result of parsing the notified context data events. Data within the batch is classified by destination, and in the end, a destination specifies the MongoDB collection where the data is going to be persisted. Thus, each destination is iterated in order to compose a per-destination data string to be persisted thanks to any `MongoBackend` implementation.
```
Comment (pcoello25) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: contanins -> contains

Comment (frbattid) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/868f39a587310f3ac46e88aa439fbdf01f9a6a90

File: doc/flume_extensions_catalogue/orion_sth_sink.md (hunk 1 above)
Comment (frbattid) [inline doc/flume_extensions_catalogue/orion_sth_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/868f39a587310f3ac46e88aa439fbdf01f9a6a90

**Comment by pcoello25** [comment]
LGTM

**Pull request merged**